scripts/compress_images_test.py

```diff
@@ -0,0 +1,250 @@
+
+        with self.swap(shutil, 'rmtree', mock_rmtree):
+            with self.swap(logging, 'error', mock_log_error):
+                with self.swap(subprocess, 'run', mock_subprocess_run):
+                    result_images: List[CompressedImageInfo] = [
+                        {
+                            'path': pathlib.Path(self.png_path),
+                            'original_size': 1000,
+                            'new_size': 500
+                        }
+                    ]
+                    compressor.compress_images(result_images)
+
+        mock_rmtree.assert_called_once_with(self.output_dir, ignore_errors=True)
+
+        mock_log_error.assert_called_with(
+            '[ERROR]: %s occurred on file %s',
+            'Permission denied error',
+            pathlib.Path(self.png_path)
+        )
+        args = [
+            'gm', 'convert',
+            str(self.png_path),
+            '-strip',
+            '-compress', 'Zip',
+            mock.ANY
+        ]
+        mock_subprocess_run.assert_called_once_with(
+            args,
+            capture_output=True,
+            text=True,
+            check=True
+        )
+
+    def test_run_no_compressible_images(self) -> None:
+        """Test run method when no images are compressible."""
+        temp_dir = tempfile.mkdtemp()
+        with open(self.txt_image_path, 'w', encoding='utf-8') as f:
+            f.write('This is not a valid image.')
+
+        compressor = compress_images.ImageCompressor(temp_dir)
+        result = compressor.run()
+        self.assertEqual(result, 0)
+
+    def test_run_with_compressible_images(self) -> None:
+        """Test run method with compressible images."""
+
+        compressor = compress_images.ImageCompressor(
+            self.temp_dir,
+            self.output_dir
+        )
+        result = compressor.run()
+        self.assertEqual(result, 1)
+
+    def test_unsupported_file_extensions(self) -> None:
+        """Test handling of unsupported file extensions."""
+        with open(self.txt_image_path, 'w', encoding='utf-8') as f:
+            f.write('Not an image')
+
+        compressor = compress_images.ImageCompressor(self.temp_dir)
+        compressible_images = compressor.find_compressible_images()
+        self.assertEqual(len(compressible_images), 1)
+
+    def test_compress_images_handles_compression_failure(self) -> None:
+        """Test handling of compression failure for an image."""
+        result_images: List[CompressedImageInfo] = [
+            {
+                'path': pathlib.Path(self.png_path),
+                'original_size': 1000,
+                'new_size': 500
+            }
+        ]
+        compressor = compress_images.ImageCompressor(
+            self.temp_dir,
+            output_dir=self.output_dir
+        )
+        with (
+            mock.patch('subprocess.run') as mock_subprocess_run,
+            mock.patch('logging.error') as mock_log_error
```

Comment:
As I mentioned in an earlier review, please use the `self.swap` family of functions instead of `mock.patch`